### PR TITLE
[DOCS] Reformat put mapping API docs

### PR DESCRIPTION
**Pull request by jrodewig**
This PR updates the put mapping API to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Relates to elastic/docs#937 and #43765

Dependent on #45699. CI failures are expected until that PR is merged.

### Preview
http://elasticsearch_45709.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/indices-put-mapping.html

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/indices/put-mapping.asciidoc [hunk 1]
```diff
@@ -75,11 +62,112 @@ you only want to rename a field, consider adding an <<alias, `alias`>> field.
...
+// TEST[setup:twitter]
+
+<1> Note that the indices specified (`twitter-1,twitter-2`) follows <<multi-index,multiple index names>> and wildcard format.
+
 For example:
 
 [source,js]
 -----------------------------------
-PUT my_index <1>
+PUT /my_index <1>
```
Comment (mayya-sharipova) [inline docs/reference/indices/put-mapping.asciidoc]: It is strange to have "For example:" sentence on line 166 and an example below in this section of the document. Before it was showing an example for updating mapping of the existing documents. Now it ended up in the section "Multiple indices".  I think we need to move these examples under the section "Restrictions for existing fields"

Comment (jrodewig) [reply docs/reference/indices/put-mapping.asciidoc]: Thanks for catching this @mayya-sharipova. It looks like I mangled the attempt to move this under the description section.

With35e9d82, I moved info about updating existing mappings back to the examples section.

**Review by mayya-sharipova — APPROVED** [review]
thanks @jrodewig. LGTM

**Pull request merged**